### PR TITLE
Add geomEach meta function

### LINKED ISSUE #12: Port Meta Tests and Functions
# *each
- [x] coordEach (@tobrun)
- [x] featureEach (#24)
- [x] flattenEach (#48 )
- [x] clusterEach (WIP @lukas-h #69)
- [x] segmentEach (WIP @armantorkzaban #70)
- [x] geomEach (#13)
- [x] propEach (#24)

# *reduce
- [x] featureReduce (WIP #49)
- [x] flattenReduce (WIP @armantorkzaban #49)
- [x] clusterReduce (WIP @lukas-h #69)
- [x] coordReduce (WIP @armantorkzaban #49)
- [x] propReduce (WIP #49)
- [x] geomReduce (WIP #49)
- [x] segmentReduce (WIP @armantorkzaban #70)

# other
- [x] coordAll (@armantorkzaban #64)
- [x] getCluster (@lukas-h #69)
- [x] ~~getType (@lukas-h)~~ [--](https://github.com/dartclub/turf_dart/issues/61#issuecomment-1062004895)
- [x] ~~getGeom (@lukas-h)~~ [--](https://github.com/dartclub/turf_dart/issues/61#issuecomment-1062004895)
- [x] getCoord (@armantorkzaban)
- [x] getCoords (@armantorkzaban)

### PR DESCRIPTION
**Pull request by baparham**
Implements the `geomEach` meta function.

Relates to #12

### CODE REVIEW

**Comment by tobrun** [comment]
@baparham what is remaining to land this PR? It's incredible that you willing to pick up the whole meta section (I'm happy to pick up some of the work though) but it might be better to land integrations 1 by 1?

**Comment by baparham** [comment]
@tobrun It was rather arbitrary to lump these all together, so perhaps progress can be faster if it's slimmed down to just `geomEach` and new PRs/issues are created to work the remaining meta functions. Perhaps tracking the overall completion of them in one top level issue or something?

File: lib/src/meta.dart [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+typedef GeomEachCallback = dynamic Function(
+  Geometry currentGeometry,
+  int featureIndex,
+  Map<String, dynamic> featureProperties,
+  BBox featureBBox,
+  dynamic featureId,
+);
+
+/// Iterate over each geometry in any GeoJSON object, similar to Array.forEach()
+void geomEach(dynamic geoJSON, GeomEachCallback callback) {
```
Comment (tobrun) [inline lib/src/meta.dart:12]: When comparing the dartdoc with the [JS equivalent](https://github.com/Turfjs/turf/blob/master/packages/turf-meta/index.js#L486-L506) some documentation is lost. While dartdoc doesn't support documentation annotations, it's still a best practice to document the parameters (see https://dart.dev/guides/language/effective-dart/documentation#do-use-prose-to-explain-parameters-return-values-and-exceptions). That doc also showcases how to use example code with using backticks.

**Review by tobrun — APPROVED** [review]
small nit on docs. Great work @baparham!

File: lib/src/meta.dart (hunk 1 above)
Comment (baparham) [reply lib/src/meta.dart:28]: I pushed a separate commit, do you think the updated docs work?

**Pull request merged**